### DISCUSSION TITLE
Execute Command "vscode.openFolder" without reloading the workspace.

### DISCUSSION THREAD

**Discussion by sayali0891** [microsoft/vscode-discussions#1263] [Extension Development QnA] [answered]
![image](https://github.com/microsoft/vscode-discussions/assets/37478397/f4f2bc67-7aad-4f2b-a1ca-acaebc8ead3a)
 I have created a **Webview** with upload file button and onClick, I am executing command "vscode.openFolder". But after executing the command, VS code reloads for the 1st time. Because of this, the execution of my custom functions is discarded.
Same as somewhat shown in this video : https://stackoverflow.com/questions/55261750/how-do-i-open-or-close-a-folder-without-reloading-the-entire-visual-studio-code 
How to stop reloading VS code for the 1st time and directly open all files present in the uploaded folder ? Also suggest if we can implement this in some other way ? Note: Uploading files/folder is my main functionality in the extension, hence need a solution desperately.

Also, are there any other commands for opening both files and folders ? (openFileFolder ie Ctrl + O does not work in the extension). Please suggest some other alternative if possible ? 
Please let me know if you need anything else from my side.

**Comment by alefragnani** [comment, marked as answer]
Hi @sayali0891 ,

Whenever you open a folder in a single-root workspace, or add a new folder on a multi-root workspace, the extensions will be _activated_ again, because the the extensions follows a _per folder_ activation mechanism. As you can follow in the SO discussion, the issue it refers is still open https://github.com/microsoft/vscode/issues/35109, with a few blockers left. I would suggest you to ping on that issue, looking for updates.

About the command to open files or folders, there is a subtle difference if you are using Windows vs MacOS. Windows does support `openFile` and `openFolder` command, but MacOS only has `openFileFolder`. You can validate the variations looking at `Preferences: Open Default Keyboard Shortcuts (JSON)` command. I suppose Linux works similar to Windows on this.

Hope this helps
